src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/client/CommunityApiClient.kt

```diff
@@ -43,4 +49,23 @@ class CommunityApiClient(
   fun getDocuments(crn: String) = getRequest<GroupedDocuments> {
     path = "/secure/offenders/crn/$crn/documents/grouped"
   }
+
+  fun getDocument(crn: String, documentId: String, outputStream: OutputStream): ClientResult<Unit> {
+    val path = "/secure/offenders/crn/$crn/documents/$documentId"
+
+    return try {
+      val bodyDataBuffer = webClient.get().uri(path)
+        .retrieve()
+        .bodyToFlux(DataBuffer::class.java)
+
+      DataBufferUtils.write(bodyDataBuffer, outputStream)
+        .share().blockLast()
```

Comment:
I'm not quite sure I follow what's going on here, but I know streaming stuff is hard to do in any language!

Reply:
I've updated the commit message with some narrative that hopefully sheds some light on this